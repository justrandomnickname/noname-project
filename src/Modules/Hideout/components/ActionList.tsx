import React from 'react'
import UI from '@UI/index'
import { ActionListElement, ActionListContainer, ActionListWrapper } from './styled/index'
import { Action } from '@Modules/Location/Action'

interface IProps {
  actions: Action.IAction[]
}

const ActionList: React.FC<IProps> = (props: IProps) => {
  return (
    <ActionListContainer>
      <ActionListWrapper>
        {props.actions.map((action, index) => (
          <ActionListElement
            isActive={action._isActive}
            onClick={() => (action._isActive ? action._callback() : () => {})}
            color={action._color}
            key={index}
          >
            <img src={action._pic} />
            <UI.GrandText>{action._text}</UI.GrandText>
          </ActionListElement>
        ))}
      </ActionListWrapper>
    </ActionListContainer>
  )
}

export default ActionList
