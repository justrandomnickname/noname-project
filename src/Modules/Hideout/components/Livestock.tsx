import React from 'react'
import UI from '@UI/index'
import { LivestockElement, LivestockContainer } from './styled/index'
import { Livestock } from '@Modules/Location/Livestock'

interface IProps {
  resources: Livestock.IResource[]
}

const LivestockComponent: React.FC<IProps> = (props: IProps) => (
  <LivestockContainer>
    {props.resources.map((resource, index) => (
      <LivestockElement key={index}>
        <UI.RegularText>
          {resource.name}: {resource.amount}
        </UI.RegularText>
      </LivestockElement>
    ))}
  </LivestockContainer>
)

export default LivestockComponent
