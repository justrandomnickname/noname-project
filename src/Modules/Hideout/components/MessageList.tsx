import React from 'react'
import UI from '@UI/index'
import { MessageListWrapper } from './styled/index'
import { Message } from '@Modules/Location/Messages/Message'

interface IProps {
  messages: Message.IMessage[]
}

const MessageList: React.FC<IProps> = (props: IProps) => {
  return (
    <MessageListWrapper>
      <UI.Title bordered>
        <span>Hideout</span>
      </UI.Title>
      {props.messages.map(message => {
        return message.tsx
      })}
    </MessageListWrapper>
  )
}

export default MessageList
