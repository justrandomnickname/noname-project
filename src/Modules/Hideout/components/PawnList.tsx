import React from 'react'
import { Pawn } from '@Modules/Pawn/Pawn'
import { PawnListElement, PawnListWrapper } from './styled/index'
import { ActionsController } from '@Controllers/ActionsController'
import UI from '@UI/index'

interface IProps {
  pawns: Pawn[]
  actionController: ActionsController
  playerInAction?: boolean
}

const PawnList: React.FC<IProps> = (props: IProps) => {
  return (
    <PawnListWrapper>
      <UI.Title bordered>
        <span>Present in area</span>
      </UI.Title>
      {props.playerInAction ? (
        <PawnListElement onClick={() => {}}>
          <UI.RegularText>Player</UI.RegularText>
          <img src="inspection.svg" />
        </PawnListElement>
      ) : null}
      {props.pawns.map(pawn => (
        <PawnListElement onClick={() => props.actionController.Dialogue('start_conversation', pawn)} key={pawn.key}>
          <UI.RegularText hoverable>{pawn.alias}</UI.RegularText>
          <img src="inspection.svg" />
        </PawnListElement>
      ))}
    </PawnListWrapper>
  )
}

export default PawnList
