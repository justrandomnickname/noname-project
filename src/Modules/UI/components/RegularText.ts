import styled from 'styled-components'

export const RegularText = styled('span')<{ hoverable?: boolean }>`
  color: ${props => props.theme.standardTextColor} !important;
  font-size: ${props => props.theme.fontSizeStandard} !important;
  line-height: 1.25em;
  font-family: 'Open Sans', regular;
  transition: 0.2s all;
  &:hover {
    cursor: ${props => (props.hoverable ? 'pointer' : null)}
    color: ${props => (props.hoverable ? props.theme.secondaryColor : props.theme.standardTextColor)} !important;
  }
`

export const GrandText = styled.span`
  font-size: 15pt;
  font-family: 'Kurale', serif;
  font-weight: 500;
  margin: 0;
  padding: 0;
`
