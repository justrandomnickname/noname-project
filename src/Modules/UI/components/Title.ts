import styled from 'styled-components'

export const Title = styled('h1')<{ bordered?: boolean }>`
  margin: 0em 0em 0.5em 0em
  padding: 0em 0em 0.25em 0em
  color: ${props => props.theme.secondaryColor};
  font-size: ${props => props.theme.fontSizeTitle};
  font-family: 'Kurale', serif;
  letter-spacing: ${props => props.theme.letterSpacingTitle};
  span {
    border-bottom: ${props => (props.bordered ? '2px ' + 'solid ' + props.theme.secondaryColor : null)};
  }
`

export const BigTitle = styled.h2`
  margin: 0;
  padding: 0.7em;
  color: ${props => props.theme.standardTextColor};
  font-size: ${props => props.theme.fontSizeBigTitle};
  font-family: 'Kurale', serif;
`
