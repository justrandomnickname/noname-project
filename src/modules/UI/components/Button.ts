import styled from 'styled-components'

export const Button = styled.button`
  margin: 0px auto;
  padding: 0.5em;
  color: ${props => props.theme.primaryColor}
  font-family: 'Kurale', serif;
  border: 0px solid black;
  background: ${props => props.theme.secondaryColor};
  box-shadow: 0px 0px 35px 1px rgba(249, 211, 66, 0.45);
  font-size: 1.5em;
  outline: 0;
  border-radius: 2px;
  line-height: 30px;
  text-transform: uppercase;
`
