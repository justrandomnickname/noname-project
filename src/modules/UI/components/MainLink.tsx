import styled from 'styled-components'
import { Link } from 'react-router-dom'
import React from 'react'
import { Props } from '@UI/interfaces/MainLink'
import UI from '@Core/UI'

const StyledLinkWrapper = styled.div`
  font-size: 1.3em;
`
// const LinkText = styled.span`
//   color: ${props => props.theme.secondaryColor};
// `

const MainLink: React.FC<Props> = (props: Props) => (
  <StyledLinkWrapper>
    <Link to={props.to}>
      <UI.Title>{props.text}</UI.Title>
    </Link>
  </StyledLinkWrapper>
)

export default MainLink
