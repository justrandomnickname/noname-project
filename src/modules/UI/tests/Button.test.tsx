import React from 'react'
import renderer from 'react-test-renderer'

import 'jest-styled-components'
import { Button } from '../components/Button'
// import styles from 'modules/UI/scss/variables.scss'

// it('(BUTTON) renders correctly', () => {
//   const wrapper = renderer.create(<Button theme={styles} />).toJSON()
//   expect(wrapper).toMatchSnapshot()
//   expect(wrapper).toHaveStyleRule('background', 'b')
// })
