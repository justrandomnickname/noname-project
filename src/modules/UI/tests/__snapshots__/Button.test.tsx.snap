// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(BUTTON) renders correctly 1`] = `
.c0 {
  width: 200px;
  height: 200px;
  border: 2px solid black;
  background: b;
}

<div
  className="c0"
/>
`;
